doc/nrf/gs_modifying.rst

```diff
@@ -14,27 +14,53 @@ After programming and testing an application, you probably want to make some mod
-This means that you must add all other files that you are using.
+The |NCS| build system is based on Zephyr, whose build system is based on `CMake`_.
+For more information, see :ref:`zephyr:application` in the Zephyr documentation.
 
-You can also configure compiler options, application defines, or include directories, or set :ref:`build types <gs_modifying_build_types>` in :file:`CMakeLists.txt`.
+In the |NCS|, the application is a CMake project.
+As such, the application controls the configuration and build process of both itself, Zephyr, and sourced libraries.
+The application's :file:`CMakeLists.txt` file is the main CMake project file and the source of the build process configuration.
+
+Zephyr creates a CMake component that must be imported by the application into its :file:`CMakeLists.txt` file.
+When imported, the application is able to reference items provided by both Zephyr and the |NCS|.
+
+Importing the Zephyr `CMake`_ component creates the ``app`` library.
+All files that your application uses must be added to this library.
+This is done directly in the :file:`CMakeLists.txt` file or files that are sourced by it.
+
+The application has full control over the build process it can impact compiler options, application defines, or include directories.
+It can also define how and which libraries are linked with the ``app`` library using either standard CMake commands or commands provided by Zephyr's CMake component.
+
+.. note::
+   Zephyr provides a :kconfig:`CONFIG_COMPILER_OPT` option to add compiler flags visible during entire build process.
 
 To update the :file:`CMakeLists.txt` file, either edit it directly or use |SES| (SES) to maintain it.
 
 Editing CMakeLists.txt directly
 ===============================
 
-Add all files that your application uses to the ``target_sources`` function in :file:`CMakeLists.txt`.
-To include several files, it can be useful to specify them with a wildcard.
-For example, to include all :file:`.c` files from the :file:`src` folder, add the following lines to your :file:`CMakeLists.txt`::
+You must add all the files that your application uses to the ``app`` library created by Zephyr.
+This is done by the :c:func:`target_sources` function provided by CMake.
+You can use it directly in :file:`CMakeLists.txt`.
 
-   FILE(GLOB app_sources src/*.c)
-   target_sources(app PRIVATE ${app_sources})
+Pay attention to the following configuration options:
 
-Instead of specifying each file (explicitly or with a wildcard), you can include all files from a folder by adding that folder as include folder::
+* If your application is complex, you may wish to split it into subdirectories.
+  In such case, these subdirectories can provide own :file:`CMakeLists.txt` files.
+* To include several files, you can specify them with a wildcard.
+  For example, to include all :file:`.c` files from the :file:`src` folder, add the following lines to your :file:`CMakeLists.txt`:
 
-   target_include_directories(app PRIVATE src)
```

Comment:
Looks the old docs were outright wrong.

Oops.

Reply:
not the first place.
I expect there are mode. I actually would like to go see all of it at some point :/